tests/tests.fish

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env fish
+
+set BASE_DIR (dirname (status -f))
+
+function test_program
+    set TEST_SCRIPT $argv[1]
+    set TEST_MODE $argv[2]
+    set TEST_ARG $argv[3]
+    set TEST_EXPECTED $argv[4]
+
+    set tmpfile (mktemp)
+    echo "$TEST_SCRIPT" >"$tmpfile"
+
+    # Fish shell's '$_' variable doesn't behave the same when sourcing from a command 
+    # substitution, so setting an environment variable is a hacky workaround
+    if test "$TEST_MODE" = "source"
+        set -x IS_SOURCED true
+        set TEST_OUT (fish -c "source $tmpfile $TEST_ARG" | string collect)
+        set -e IS_SOURCED
+    else if test "$TEST_MODE" = "direct"
+        chmod +x "$tmpfile"
+        set TEST_OUT (fish -c "$tmpfile $TEST_ARG" | string collect)
+    else if test "$TEST_MODE" = "envgrep"
+        set -x IS_SOURCED true
+        set TEST_OUT (fish -c "source $tmpfile $TEST_ARG; env | grep TEST_AWS_CLI_MFA" | string collect)
+        set -e IS_SOURCED
+    else
+        echo "BAD TEST ARG!"
+        return
+    end
+
+    rm "$tmpfile"
+
+    if test "$TEST_OUT" = "$TEST_EXPECTED"
+        echo "SUCCESS - $TEST_MODE: $TEST_ARG"
+    else
+        echo "$TEST_EXPECTED"
+        echo "FAILURE - $TEST_MODE: $TEST_ARG: $TEST_OUT"
+    end
+end
+
+set FISH_SCRIPT (sed 's/"/\\\"/g' $BASE_DIR/test_sh.py | sed -e "/#INSERT_PYTHON_CODE_HERE/r /dev/stdin" -e "s///" $BASE_DIR/../src/mfa.fish | string collect)
+
+test_program (string replace "python3" "python1" $FISH_SCRIPT | string collect) source python3 "python1 could not be found!"
+test_program (string replace "python3" "python1" $FISH_SCRIPT | string collect) direct python3 "python1 could not be found!"
```

Comment:
This error message is incorrect, it should be `python3 could not be found`

Reply:
https://github.com/rogusdev/aws-cli-mfa/blob/642b9356122ba1fd2068133ac3d168a51f760f47/tests/tests.sh#L50

Reply:
Yeah this was a result of shortening the requirements check since it used the same variable we were looping through.  I can expand this back out to maintain consistency with the others.

Reply:
Ah, if that's the case, I suppose I can live with this